components/brave_wallet_ui/page/screens/composer_ui/select_token_modal/select_token_modal.tsx

```diff
@@ -167,270 +235,296 @@ export const SelectTokenModal = React.forwardRef<HTMLDivElement, Props>(
       })
     })
 
-    const { userVisibleTokensInfo } = useGetUserTokensRegistryQuery(undefined, {
-      selectFromResult: (result) => ({
-        userVisibleTokensInfo: selectAllVisibleUserAssetsFromQueryResult(result)
-      })
-    })
+    const { data: combinedTokenRegistry } = useGetCombinedTokensRegistryQuery()
 
-    const { data: fullTokenList } = useGetCombinedTokensListQuery(
-      showFullFlatTokenList ? undefined : skipToken
-    )
+    const fullVisibleTokenList = React.useMemo(() => {
+      if (!combinedTokenRegistry) {
+        return []
+      }
+      return getEntitiesListFromEntityState(
+        combinedTokenRegistry,
+        combinedTokenRegistry.visibleTokenIds
+      )
+    }, [combinedTokenRegistry])
+
+    const { userVisibleFungibleTokens, userVisibleNfts } =
+      useGetUserTokensRegistryQuery(undefined, {
+        selectFromResult: (result) => ({
+          userVisibleFungibleTokens:
+            selectAllVisibleFungibleUserAssetsFromQueryResult(result),
+          userVisibleNfts: selectAllVisibleUserNFTsFromQueryResult(result)
+        })
+      })
 
     const networks = modalType === 'swap' ? swapNetworks : visibleNetworks
 
     const { data: tokenBalancesRegistry, isLoading: isLoadingBalances } =
-      useBalancesFetcher(
-        !showFullFlatTokenList
-          ? {
-              accounts,
-              networks
-            }
-          : skipToken
-      )
+      useBalancesFetcher({
+        accounts,
+        networks
+      })
 
-    // Methods
-    const getTokenListByAccount = React.useCallback(
-      (account: BraveWallet.AccountInfo) => {
-        if (!account || !networks) {
-          return []
-        }
-        // Since LOCALHOST's chainId is shared between coinType's
-        // this check will make sure we are returning the correct
-        // LOCALHOST asset for each account.
-        const coinName = CoinTypesMap[account.accountId.coin]
-        const localHostCoins = userVisibleTokensInfo.filter(
-          (token) => token.chainId === BraveWallet.LOCALHOST_CHAIN_ID
-        )
-        const accountsLocalHost = localHostCoins.find(
-          (token) => token.symbol.toUpperCase() === coinName
-        )
+    const swapNetworksChainIds = React.useMemo(() => {
+      return swapNetworks.map((network) => network.chainId)
+    }, [swapNetworks])
 
-        const chainList = filterNetworksForAccount(
-          networks,
-          account.accountId
-        ).map((network) => network.chainId)
+    const tokensBySelectedComposerOption = React.useMemo(() => {
+      if (modalType === 'swap') {
+        return fullVisibleTokenList
+          .filter(
+            (token) => !token.isErc721 && !token.isErc1155 && !token.isNft
+          )
+          .filter((token) => swapNetworksChainIds.includes(token.chainId))
```

Comment:
perf: noticing that `swapNetworksChainIds` is only used here. I guess we can remove that variable/memo in favor of checking the chainId here

```suggestion
          .filter((token) => swapNetworks.some(({ chainId }) => chainId === token.chainId))
```

Reply:
Changed to use `.some`